SleepEngine/Engine/Utils/file_utils.cpp

```diff
@@ -0,0 +1,22 @@
+#include "stdafx.h"
+#include "file_utils.h"
+
+std::optional<std::string> readFile(std::filesystem::path path)
+{
+    if (!std::filesystem::exists(path))
+    {
+        return std::nullopt;
+    }
+
+    int bytes = static_cast<int>(std::filesystem::file_size(path));
+
+    std::string data(bytes, '#');
+    std::ifstream fstream;
+
+    assertion(!fstream.is_open(), "stats file already open");
```

Comment:
I would use something like:
`fmt::format("File {} is already opened", path)`
instead of your strange string.

Reply:
.c_str()

Reply:
it's so clumbersome
the spdlog accepts "File {} is already opened", path already, can we delegate all the parameters to it?
macros support variadic params

Reply:
have done just that

Reply:
Da fuq, it works? Niceu